packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -80,3 +93,608 @@ class CupertinoColors {
+  final Color defaultColor;
+
+  /// The color to use when the [BuildContext] implies a combination of light mode,
+  /// normal contrast, and base interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get normalColor => _colorMap[0];
+
+  /// The color to use when the [BuildContext] implies a combination of dark mode,
+  /// normal contrast, and base interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get darkColor => _colorMap[1];
+
+  /// The color to use when the [BuildContext] implies a combination of light mode,
+  /// high contrast, and base interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get highContrastColor => _colorMap[2];
+
+  /// The color to use when the [BuildContext] implies a combination of dark mode,
+  /// high contrast, and base interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get darkHighContrastColor => _colorMap[3];
+
+  /// The color to use when the [BuildContext] implies a combination of light mode,
+  /// normal contrast, and elevated interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get elevatedColor => _colorMap[4];
+
+  /// The color to use when the [BuildContext] implies a combination of dark mode,
+  /// normal contrast, and elevated interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get darkElevatedColor => _colorMap[5];
+
+  /// The color to use when the [BuildContext] implies a combination of light mode,
+  /// high contrast, and elevated interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get highContrastElevatedColor => _colorMap[6];
+
+  /// The color to use when the [BuildContext] implies a combination of dark mode,
+  /// high contrast, and elevated interface elevation.
+  ///
+  /// Defaults to [defaultColor] when unspecified.
+  Color get darkHighContrastElevatedColor => _colorMap[7];
+
+  final List<Color> _colorMap;
+
+  /// Resolves the given [Color] to a concrete [Color], using the given [BuildContext].
+  /// If the given color is already a concrete [Color], it will be returned as is.
+  static Color resolve(Color resolvable, BuildContext context, { bool nullOk = false })
+    => (resolvable is CupertinoDynamicColor)
+      ? resolvable.resolveFrom(context, nullOk: nullOk)
+      : resolvable;
+
+  /// Resolves this `CupertinoDynamicColor` using the provided [BuildContext].
+  ///
+  /// Calling this method will create a new `CupertinoDynamicColor` that is almost
+  /// identical to the orignal `CupertinoDynamicColor` except with a different
+  /// [defaultColor], such that when the new `CupertinoDynamicColor` is used as
+  /// a regular [Color], it is equivalent to one of the specified colors that
+  /// matches the given [BuildContext]'s traits.
+  ///
+  /// For example, if the given [BuildContext] indicates the widgets in the subtree
```

Comment:
The documentation here and elsewhere would be easier to understand if it avoided synonyms like "interface elevation", "vibrancy", and "accessibility contrast", and etc, and only used terms that come from the API, like [MediaQuery.platformBrightness], [MediaQuery.highContrast] (TBD) and, [CupertinoUserInterfaceLevel].